packages/alert-banner/src/AlertBanner.ts

```diff
@@ -0,0 +1,146 @@
+import styles from './alert-banner.css.js';
+
+export type AlertBannerVariants = 'neutral' | 'info' | 'negative' | '';
+const VALID_VARIANTS = ['neutral', 'info', 'negative'];
+
+/**
+ * @element sp-alert-banner
+ */
+export class AlertBanner extends SpectrumElement {
+    public static override get styles(): CSSResultArray {
+        return [styles];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @property({ type: Boolean, reflect: true })
+    public dismissible = false;
+
+    /**
+     * The variant applies specific styling when set to `negative` or `info`;
+     * `variant` attribute is removed when it's passed an invalid variant.
+     *
+     * @param {String} variant
+     */
+    @property({ type: String })
+    public set variant(variant: AlertBannerVariants) {
+        if (variant === this.variant) {
+            return;
+        }
+        const oldValue = this.variant;
+        if (this._isValidVariant(variant)) {
+            this.setAttribute('variant', variant);
+            this._variant = variant;
+        } else {
+            this.removeAttribute('variant');
+            this._variant = '';
+        }
+        this.requestUpdate('variant', oldValue);
+    }
+
+    public get variant(): AlertBannerVariants {
+        return this._variant;
+    }
+
+    private _variant: AlertBannerVariants = '';
+
+    protected _isValidVariant(variant: string): boolean {
+        return VALID_VARIANTS.includes(variant);
+    }
+
+    protected renderIcon(variant: string): TemplateResult {
+        switch (variant) {
+            case 'info':
+                return html`
+                    <sp-icon-info
+                        label="Information"
+                        class="type"
+                    ></sp-icon-info>
+                `;
+            case 'negative':
+                return html`
+                    <sp-icon-alert label="Error" class="type"></sp-icon-alert>
+                `;
+            default:
+                return html``;
+        }
+    }
+
+    private _shouldClose(): void {
+        const applyDefault = this.dispatchEvent(
+            new CustomEvent('close', {
+                composed: true,
+                bubbles: true,
+                cancelable: true,
+            })
+        );
+        if (applyDefault) {
+            this.close();
+        }
+    }
+
+    public close(): void {
+        this.open = false;
+    }
+
+    private _handleKeydown(event: KeyboardEvent): void {
+        if (event.code === 'Escape' && this.dismissible) {
+            this._shouldClose();
+        }
+    }
+
+    protected override render(): TemplateResult {
+        return html`
+            <div class="body" role="alert">
+                <div class="content">
+                    ${this.renderIcon(this.variant)}
+                    <div class="text"><slot></slot></div>
+                </div>
+                <slot name="action"></slot>
```

Comment:
Can you surface this slot to the description of the component so that @swc-react can also leverage this and also cascades to the API section of the documentation